packages/jest-core/src/runJest.ts

```diff
@@ -192,6 +192,9 @@ export default async function runJest({
     }),
   );
 
+  allTests = globalConfig.shard
+    ? sequencer.shard(allTests, globalConfig.shard)
```

Comment:
this should also use `await` I think, in case people want to do some async work to figure out their strategy (same as `sort` today).